index.js

```diff
@@ -0,0 +1,213 @@
+'use strict';
+
+var thrift = require('thrift');
+var through = require('through2');
+var Types = require('./src/bolt_types');
+var BoltManagerService = require('./src/BoltManagerService');
+var BoltProxyService = require('./src/BoltProxyService');
+var BoltSchedulerService = require('./src/BoltSchedulerService');
+
+var ComputationMetadata = Types.ComputationMetadata;
+var StreamMetadata = Types.StreamMetadata;
+//
+// Transation returned on each request
+//
+var Tx = Types.ComputationTx;
+var Record = Types.Record;
+var Endpoint = Types.Endpoint;
+
+
+function Computation(options) {
+  //
+  // Remark: Do we want to fail if we dont get a name? For now lets generate
+  // a unique identifier
+  //
+  this.name = options.name || 'node-' + process.hrtime().join('-');
+  //
+  // Set the grouping to the correct INT code based on what we have in the
+  // thrift types. We default to SHUFFLE
+  //
+  this.grouping = groupMap(options.grouping || 'shuffle');
+
+  if (!this.grouping) throw new Error('Invalid grouping ' + options.grouping);
+
+  //
+  // Computation init function and process functions which are required
+  //
+  this._init = typeof options.init === 'function' ? options.init : undefined;
+  this._processRecord = typeof options.processRecord === 'function'
+    ? options.processRecord
```

Comment:
we should make a comment that we validate the object later for the functions required by the driver.